addons/l10n_in/models/account_invoice.py

```diff
@@ -36,17 +36,19 @@ def _compute_l10n_in_gst_treatment(self):
         for record in self:
             record.l10n_in_gst_treatment = record.partner_id.l10n_in_gst_treatment
 
-    @api.model
-    def _l10n_in_get_indian_state(self, partner):
-        """In tax return filing, If customer is not Indian in that case place of supply is must set to Other Territory.
-        So we set Other Territory in l10n_in_state_id when customer(partner) is not Indian
-        Also we raise if state is not set in Indian customer.
-        State is big role under GST because tax type is depend on.for more information check this https://www.cbic.gov.in/resources//htdocs-cbec/gst/Integrated%20goods%20&%20Services.pdf"""
-        if partner.country_id and partner.country_id.code == 'IN' and not partner.state_id:
-            raise ValidationError(_("State is missing from address in '%s'. First set state after post this invoice again.", partner.name))
-        elif partner.country_id and partner.country_id.code != 'IN':
-            return self.env.ref('l10n_in.state_in_ot')
-        return partner.state_id
+    @api.depends('partner_id')
+    def _compute_l10n_in_state_id(self):
+        for move in self:
+            if move.country_code == 'IN':
+                country_code = move.partner_id.country_id.code
+                if country_code == 'IN':
+                    move.l10n_in_state_id = move.partner_id.state_id
+                elif country_code:
+                    move.l10n_in_state_id = self.env('l10n_in.state_in_oc')
+                else:
+                    move.l10n_in_state_id = move.company_id.state_id
```

Comment:
Why should it be any different than line 45?

Reply:
if there is no partner then no country_code for POS entry so we set the company state as POS

Reply:
But maybe the state of the partner is Indian...  If not, then he should have supplied a country.

Reply:
```suggestion
                    move.l10n_in_state_id = move.state_id.country_id.code == "IN" and move.state_id or move.company_id.state_id
``` Bu testing in the interface the country is automatically set when you select the state...